Add wrapInput option.

williamboman opened the pull request:
Make it optional to wrap the input or not. Fixes these kind of issues;

**Before**
![Before](http://i.imgur.com/STV57hB.png)

**After**
![After](http://i.imgur.com/TrVCbEY.png)

williamboman committed:
Add wrapInput option.

williamboman committed:
Add wrapInput option to index.html.

williamboman commented on index.html:
```diff
@@ -1,19 +1,19 @@
 <html>
 	<head>
-	
+
```
My editor did these (remove trailing whitespace) :expressionless:

williamboman commented:
Ping @erming :)

erming commented:
Hey, sorry for the delay!
What causes the issue for you? Is it the `position: relative;` that screws your example over?

I'll go ahead and merge.

williamboman commented:
Well basically the whole wrapping of the `input` with a `div` element caused it.

[I wrote a quick, very ugly, demo.](http://codepen.io/anon/pen/afvEy)

williamboman commented:
By the way, could you tag a new release on the latest ref, would like to use this in production :)

erming commented:
Will do!

erming commented:
Ah, now I see what caused the problem. You're expecting it to work like an inline element. It seems using `display: inline-block;` on the wrapper solves this. I'll experiment with that later (might make it the default behavior).

williamboman commented:
Yeah. Now when I think about it my fix is actually bad. It should definitely wrap the input, but the wrap element should use `inline-block` in order to imitate the default positioning of an `input` element.

Perhaps dynamically set the `display` CSS property of the wrap element by looking at the `input`?

erming commented:
If I remember correctly, I had to use a wrapper to position the clone exactly behind the input.

I'm all for getting it to work without the need of any settings. I think that sounds like a good idea!

williamboman commented:
Yeah, I didn't really think it through. [I reverted my changes and added this.](https://github.com/williamboman/tabcomplete/blob/gh-pages/src/tabcomplete.js#L217)

erming commented:
Hmm.. That works on the tabcomplete demo `index.html`, but I think input.css("block"); will default to undefined sometimes

erming commented:
No, not undefined. But this was my result:

Defaults:
- Chrome, Internet Explorer: `inline-block`
- Firefox: `inline`

Try this one: http://jsbin.com/pujodaloho/1/edit?html,js,output

williamboman commented:
Do you mean `input.css('display')`? Afaik it will always resolve to the browser defaults (`inline-block` on virtually every browser) if there are no rulesets for the element. Browsers with no default properties will default to `undefined`, I suppose.

williamboman commented:
In the end it will still imitate the `display` property properly, which is the wanted outcome. I think jQuery doesn't set the property if the value is `undefined`.

erming commented:
I just released `1.5.0`

Time to slack in front of the TV.
Have a good one! :)

williamboman commented:
Nice, thanks!
